hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -112,6 +112,9 @@ public S3Auth getSignature() {
   @NotNull
   @VisibleForTesting
   OzoneClient createOzoneClient() throws IOException {
+    // S3 Gateway should always set the S3 Auth. OM can choose to ignore it
+    // based on the security configuration.
```

Comment:
Can you explain what this comment means?
OM can choose to ignore it
    // based on the security configuration.

Reply:
S3 Clients will always set the authentication information as S3 APIs do not really have a contract for unauthenticated communication. Thus, it is best for S3G to forward the signature to OM. OM can choose to validate the signature or not based on how OM is configured.

Reply:
Essentially do not skip setting S3 auth on the client side even if security is not enabled.

Reply:
Got it. But the 2nd part in comment is confusing for readers.

Reply:
Addressed, dropped the OM handling bit.